### PR TITLE
[sonic_platform/sfp_base] Add common definition for get SFP error status

### PR DESCRIPTION
**Pull request by stephenxs**
#### Description

Add common definitions for getting SFP error status

- Platform API definition: `get_sfp_status`
- Related constant definitions

Signed-off-by: Stephen Sun <stephens@nvidia.com>

### CODE REVIEW

File: sonic_platform_base/sfp_base.py [hunk 1]
```diff
@@ -16,6 +16,41 @@ class SfpBase(device_base.DeviceBase):
...
 
+    # Generic error types definition
+    SFP_STATUS_INITIALIZING                         = 'Initializing'
+    SFP_STATUS_OK                                   = 'OK'
+    SFP_STATUS_UNPLUGGED                            = 'Unplugged'
+    SFP_STATUS_DISABLED                             = 'Disabled'
+    SFP_ERROR_DESCRIPTION_BLOCKING                  = 'Blocking EEPROM from being read'
+    SFP_ERROR_DESCRIPTION_POWER_BUDGET_EXCEEDED     = 'Power budget exceeded'
+    SFP_ERROR_DESCRIPTION_I2C_STUCK                 = 'Bus stuck (I2C data or clock shorted)'
+    SFP_ERROR_DESCRIPTION_BAD_EEPROM                = 'Bad or unsupported eeprom'
```
Comment (jleveque) [inline sonic_platform_base/sfp_base.py:27]: s/eeprom/EEPROM/

**Review by jleveque — CHANGES_REQUESTED** [review]

File: sonic_platform_base/sfp_base.py (hunk 1 above)
Comment (stephenxs) [reply sonic_platform_base/sfp_base.py]: Fixed.

**Review by jleveque — APPROVED** [review]

**Comment by stephenxs** [comment]
Looks like it failed due to an unsatisfied dependency on yang module
```
2021-06-14T17:16:15.4207828Z WARNING: Running pip as root will break packages and permissions. You should install packages reliably by using venv: https://pip.pypa.io/warnings/venv
2021-06-14T17:16:15.4660349Z + sudo pip3 install sonic_config_engine-1.0-py3-none-any.whl
2021-06-14T17:16:15.8202969Z Processing ./sonic_config_engine-1.0-py3-none-any.whl
2021-06-14T17:16:15.8991885Z ERROR: Could not find a version that satisfies the requirement sonic-yang-mgmt>=1.0 (from sonic-config-engine) (from versions: none)
2021-06-14T17:16:15.8992828Z ERROR: No matching distribution found for sonic-yang-mgmt>=1.0
```

**Comment by jleveque** [comment]
> Looks like it failed due to an unsatisfied dependency on yang module
> 
> ```
> 2021-06-14T17:16:15.4207828Z WARNING: Running pip as root will break packages and permissions. You should install packages reliably by using venv: https://pip.pypa.io/warnings/venv
> 2021-06-14T17:16:15.4660349Z + sudo pip3 install sonic_config_engine-1.0-py3-none-any.whl
> 2021-06-14T17:16:15.8202969Z Processing ./sonic_config_engine-1.0-py3-none-any.whl
> 2021-06-14T17:16:15.8991885Z ERROR: Could not find a version that satisfies the requirement sonic-yang-mgmt>=1.0 (from sonic-config-engine) (from versions: none)
> 2021-06-14T17:16:15.8992828Z ERROR: No matching distribution found for sonic-yang-mgmt>=1.0
> ```

It appears this is a new dependency which broke the CI. I have opened a PR here in an attempt to fix: https://github.com/Azure/sonic-platform-common/pull/198

**Pull request merged**